sdks/java/io/hadoop-file-system/src/main/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystemRegistrar.java

```diff
@@ -45,6 +45,10 @@
 
   private static final List<String> HA_SCHEMES = Arrays.asList("hdfs", "webhdfs");
 
+  // Using hard-coded value for compatibility, because hadoop-dfs-client:2.7's DFSConfigKeys holds
+  // this value while it is HdfsClientConfigKeys that holds the value in version 2.8.
+  private static final String CONFIG_KEY_DFS_NAMESERVICES = "dfs.nameservices";
```

Comment:
Updated the source to use a string rather than a field of Hadoop class.

Reply:
Yes you read my mind, I was going to subject that just as the best path forward.

Reply:
I think this fix makes this mergeable, do you plan to tackle the multi version testing as well?

Reply:
@iemejia Thanks. Would you merge this?

I don't add the test in this PR. I created a followup issue: https://issues.apache.org/jira/browse/BEAM-8876